contracts/airdrop/StakingAirdrop.sol

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.30;
+
+/// @dev Service Registry interface
+interface IServiceRegistry {
+    /// @dev Gets the owner of a specified service Id.
+    /// @param serviceId Service Id.
+    /// @return serviceOwner Service owner address.
+    function ownerOf(uint256 serviceId) external view returns (address serviceOwner);
+}
+
+/// @dev Generic IERC20 token interface
+interface IToken {
+    /// @dev Gets the amount of tokens owned by a specified account.
+    /// @param account Account address.
+    /// @return Amount of tokens owned.
+    function balanceOf(address account) external view returns (uint256);
+
+    /// @dev Transfers the token amount.
+    /// @param to Address to transfer to.
+    /// @param amount The amount to transfer.
+    /// @return True if the function execution is successful.
+    function transfer(address to, uint256 amount) external returns (bool);
+}
+
+/// @dev Provided zero address.
+error ZeroAddress();
+
+/// @dev Provided zero value.
+error ZeroValue();
+
+/// @dev Only `owner` has a privilege, but the `sender` was provided.
+/// @param sender Sender address.
+/// @param owner Required sender address as an owner.
+error OwnerOnly(address sender, address owner);
+
+/// @dev Wrong length of two arrays.
+/// @param numValues1 Number of values in a first array.
+/// @param numValues2 Number of values in a second array.
+error WrongArrayLength(uint256 numValues1, uint256 numValues2);
+
+/// @dev Provided incorrect amount.
+/// @param provided Provided amount.
+/// @param expected Expected amount.
+error WrongAmount(uint256 provided, uint256 expected);
+
+/// @dev Value overflow.
+/// @param provided Overflow value.
+/// @param max Maximum possible value.
+error Overflow(uint256 provided, uint256 max);
+
+/// @dev Caught reentrancy violation.
+error ReentrancyGuard();
+
+
+/// @title StakingAirdrop - Smart contract for staking airdrop
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
+contract StakingAirdrop {
+    event Claimed(address indexed sender, uint256 indexed serviceId, uint256 amount);
+
+    // Version number
+    string public constant VERSION = "0.1.0";
+    // Airdrop amount
+    uint256 public constant AIRDROP_AMOUNT = 10_000;
```

Comment:
what token we will use?  We already know that the amount is what we will distribute at max? Are we considering the right decimals?

Reply:
This is just a mock, we'll have to pass one when finalized.

Reply:
Also, not sure about the token.